File: passenger-service/src/main/java/io/simakkoi9/passengerservice/model/converter/UserStatusConverter.java
package io.simakkoi9.passengerservice.model.converter;

import io.simakkoi9.passengerservice.model.entity.UserStatus;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter(autoApply = true)
public class UserStatusConverter implements AttributeConverter<UserStatus, Integer> {

    @Override
    public Integer convertToDatabaseColumn(UserStatus userStatus) {
        if (userStatus == null) {
            return null;
        }
        return userStatus.getCode();
    }

    @Override
    public UserStatus convertToEntityAttribute(Integer code) {
        if (code == null) {
            return null;
        }
        return UserStatus.fromCode(code);
    }
}

File: driver-service/src/main/kotlin/io/simakkoi9/driverservice/model/converter/UserStatusConverter.kt
package io.simakkoi9.driverservice.model.converter

import io.simakkoi9.driverservice.model.entity.UserStatus
import jakarta.persistence.Converter
import jakarta.persistence.AttributeConverter

@Converter(autoApply = true)
class UserStatusConverter : AttributeConverter<UserStatus, Int> {
    override fun convertToDatabaseColumn(userStatus: UserStatus?): Int? {
        return userStatus?.code
    }

    override fun convertToEntityAttribute(integer: Int?): UserStatus? {
        return integer?.let { UserStatus.fromCode(it) }
    }
}